src/SFA.DAS.EmployerIncentives.Functions.LegalEntities/HandleEarningsResilienceCheck.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EmployerIncentives.Functions.LegalEntities.Services.LegalEntities;
+
+namespace SFA.DAS.EmployerIncentives.Functions.LegalEntities
+{
+    public class HandleEarningsResilienceCheck
+    {
+        private readonly IEarningsResilienceCheckService _earningsResilienceCheckService;
+
+        public HandleEarningsResilienceCheck(IEarningsResilienceCheckService earningsResilienceCheckService)
+        {
+            _earningsResilienceCheckService = earningsResilienceCheckService;
+        }
+
+        [FunctionName("TimerEarningsResilienceCheck")]
+        public async Task RunTimer([TimerTrigger("%EarningsResilienceCheckTriggerTime%", RunOnStartup = false)]TimerInfo myTimer, ILogger log)
+        {
+            await _earningsResilienceCheckService.Update();
+        }
+
+        [FunctionName("HttpEarningsResilienceCheck")]
+        public async Task<IActionResult> RunHttp([HttpTrigger(AuthorizationLevel.Function)] HttpRequest request)
+        {
+            await _earningsResilienceCheckService.Update();
+
+            return new OkResult();
+        }
```

Comment:
I assumed we have the RunHttp call for testing, but your not calling it from the the ATs. Can we delete this extra function?

Reply:
I thought it would be useful both for testing, and ad hoc situations where we ask DevOps to run the function. Happy to remove after testing before the merge if it's not needed.

Reply:
That's a good point, worth asking if it's possible this will be needed